File: 7. Code Challenges/6. Greatest Common Divisor (GCD) of 2 number/gcd.js


function GCD(num1,num2){
    let divisors = [];
    for (const i of Array.from(Array(Math.max(num1,num2)), (_,i)=> i+1)) {
        if (num1 % i == 0 && num2 % i == 0){
            divisors.push(i);
        }
    }
    return divisors.reverse()[0];
}

console.log(GCD(40,120)); // 40
console.log(GCD(25,50));  // 25
console.log(GCD(6,9));    // 3

File: 7. Code Challenges/6. Greatest Common Divisor (GCD) of 2 number/gcd.py


def GCD(num1, num2):
    divisors = [i for i in range(2,min(num1,num2)+1) if (num1%i==0 and num2%i==0)]
    return max(divisors)

print(f"GCD: {GCD(90,120)}") # 30
print(f"GCD: {GCD(50,100)}") # 50
print(f"GCD: {GCD(45,78)}")  # 3

